### PR TITLE
Bolded yelling and missing green span

### PR DESCRIPTION
**Pull request by ghost**
Makes text bold when you yell with double exclamation points.

Comdom exclaims, "**GET THAT FUKKEN DISK!!**"

Fixes missing 'green' span for when you successfully apply a bruise pack/ointment on yourself.

### CODE REVIEW

**Comment by Cheridan** [comment]
Man, I dunno. It's possible to exclaim like "Ah!" when startled, without **YELLING AT THE TOP OF YOUR LUNGS**. Like if it needed "!!!" at the end then that'd be cool.

**Comment by paprka** [comment]
Even exclaiming "**Ah!**", it makes sense for it to be bolded, imo.

**Comment by Iamgoofball** [comment]
**I'M TALKING LIKE THIS FROM NOW ON!!!**

**Comment by CollenN** [comment]
I don't like this. Like, not all sentences ending in an exclamation point are indications that the speaker is shouting. And if someone puts multiple sentences within a single line, this would bold all of it, not just the sentence that ends in an exclamation point.

Not to mention that people will abuse it, as Goofball has so gracefully demonstrated.

**Comment by paprka** [comment]
It's very rare that people put multiple sentences in a single line, not to mention poor form for roleplay anyway

**Comment by dannno** [comment]
There are a lot of things people could say that end with exclamation points that wouldn't need bolding
Like just being excited!
how about making it two exclamation points or more to bold

**Comment by PKPenguin321** [comment]
I like the two or more exclamation points bolding it, but not one exclamation point, for the reasons listed. It couldn't be that hard to tweak so it needs two !s, could it?

**Comment by paprka** [comment]
I don't think the current check allows for that. I'll try.

**Comment by dannno** [comment]
what about sentences that start with ! instead

**Comment by paprka** [comment]
Again, not a saycode wizard

**Comment by dannno** [comment]
i'm sure it's possible, and feels like a better solution

**Comment by paprka** [comment]
I will wait patiently for @Miauw62 to either close this because it doesn't ~fix everything~ or show me how to check for double exclamation points as endings

**Comment by Razharas** [comment]
````if(copytext(original_text, length(original_text), length(original_text)+1) == "!!")````

**Comment by paprka** [comment]
you're joking right

**Comment by Iamgoofball** [comment]
would checking for ending "!!" work?

**Comment by paprka** [comment]
tried it, didn't, i'll try razharas' suggestion

**Comment by paprka** [comment]
@Razharas That did not work

**Comment by mrpain666** [comment]
Oh I'm going to have so much fun with this

File: code/modules/mob/living/silicon/ai/say.dm [hunk 1]
```diff
@@ -25,7 +25,7 @@
 	if (ending == "?")
 		return "queries, \"<span class = 'robot'>[text]</span>\"";
 	else if (ending == "!")
-		return "declares, \"<span class = 'robot'>[text]</span>\"";
+		return "declares, \"<span class = 'robotyell'>[text]</span>\"";
```
Comment (Miauw62) [inline code/modules/mob/living/silicon/ai/say.dm]: please make this ```<span class='robot'><span class='yell'>```, that would be much more convenient for me.

**Comment by Miauw62** [comment]
razharas is a butt who can't code because the correct code would be ```copytext(text, length(text) - 1)```

File: code/modules/mob/living/silicon/ai/say.dm (hunk 1 above)
Comment (paprka) [inline code/modules/mob/living/silicon/ai/say.dm]: Aight

**Comment by paprka** [comment]
Yelling now requires double exclamation points. Robotyell removed, replaced with robot and yell span classes.

Tested and working, thanks meow.

**Comment by dannno** [comment]
:+1:

**Comment by Steelpoint** [comment]
If that's the case then :+1:

**Pull request merged**